subsys/net/lib/sockets/sockets.c

```diff
@@ -470,10 +585,56 @@ ssize_t zsock_recvfrom(int sock, void *buf, size_t max_len, int flags,
 	return 0;
 }
 
+#ifdef CONFIG_USERSPACE
+Z_SYSCALL_HANDLER(zsock_recvfrom, sock, buf, max_len, flags, src_addr,
+		  addrlen_param)
+{
+	socklen_t addrlen_copy;
+	socklen_t *addrlen_ptr = (socklen_t *)addrlen_param;
+	ssize_t ret;
+
+	if (Z_SYSCALL_OBJ(sock, K_OBJ_NET_CONTEXT)) {
+		errno = EBADF;
+		return -1;
+	}
+
+	struct net_context *ctx = INT_TO_POINTER(sock);
```

Comment:
Shouldn't these variables declared at the beginning of the func?

Reply:
Can you show me where this is mandated in our coding style? I can't find any prohibition against it and checkpatch is AFAIK happy.